agents/template/src/main/scala/org/broadinstitute/transporter/kafka/TransferState.scala

```diff
@@ -0,0 +1,46 @@
+package org.broadinstitute.transporter.kafka
+
+import io.circe.{Encoder, Json}
+import io.circe.syntax._
+import org.broadinstitute.transporter.kafka.TransferStreamBuilder.UnhandledErrorInfo
+import org.broadinstitute.transporter.transfer.TransferResult
+
+/** The possible states of a transfer request as seen by an agent. */
+sealed abstract class TransferState[+I, +P, +O] extends Product with Serializable {
+  def isDone: Boolean
+  def message: Json
+}
+
+final case class Progress[P: Encoder](value: P)
+    extends TransferState[Nothing, P, Nothing] {
+  def isDone = false
+
+  def message: Json = (0, value).asJson
```

Comment:
If we have `step` return this class too, we should put the step # in the list of parameters instead of hard-coding it to zero here.

Reply:
Though it's really more of an implementation detail, so it'd be unfortunate for each runner implementation to have to deal with it...